### PR TITLE
Find unused deps script and cleanup existing cruft

### LINKED ISSUE #42432: Script for cleaning up stale entries in version text files
There have been instances where a dependency was removed from the product but the dependency was still listed in `external_versions.txt`. Investigate enhancing the scanner to flag when listed dependencies are unused.

### PR DESCRIPTION
**Pull request by billwert**
This will find any unused lines in external_dependencies.txt and version_client.txt. It won't remove them. If it finds something it will cause a build failure.

Additionally this cleans up a handful of existing old entries that I think are no longer needed. `pom_file_version_scanner.ps1` still works, and grepping for the entries I removed didn't turn up anything that looked like it would break something.

Fixes #42432

### CODE REVIEW

**Comment by billwert** [comment]
Pipeline experience in case of a failure:
```
/opt/hostedtoolcache/Python/3.11.10/x64/bin/python /mnt/vss/_work/1/s/eng/versioning/find_unused_dependencies.py
Unused external_dependencies.txt entries:
  springboot3_org.testcontainers:junit-jupiter-nothere

##[error]The process '/opt/hostedtoolcache/Python/3.11.10/x64/bin/python' failed with exit code 1
Finishing: Verify version text files
```

**Comment by billwert** [comment]
@Netyyyy @saragluna Lots of change in the Spring dependency listings here that I think is correct. I'd very much appreciate your input!

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -119,66 +99,12 @@ testdep_net.bytebuddy:byte-buddy;1.15.5
 testdep_net.bytebuddy:byte-buddy-agent;1.15.5
 
 ## Spring boot dependency versions
-org.springframework.boot:spring-boot-dependencies;2.7.18
```
Comment (saragluna) [inline eng/versioning/external_dependencies.txt:122]: @Netyyyy, will our script regenerate and append these entries here?

Comment (Netyyyy) [reply eng/versioning/external_dependencies.txt:122]: No, we won't update and use these dependencies for Spring Boot2

**Review by saragluna — APPROVED** [review]

**Review by alzimmermsft — APPROVED** [review]

File: eng/versioning/version_data_java_files.txt [hunk 2]
```diff
@@ -3,5 +3,4 @@
 # has been fixed this file and the update logic in update_versions.py should no longer be necessary.
 
 # The file format here should be the relative path from the root of the azure-sdk-for-java
-sdk/eventhubs/microsoft-azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/impl/ClientConstants.java
 sdk/storage/microsoft-azure-storage-blob/src/main/java/com/microsoft/azure/storage/blob/Constants.java
```
Comment (alzimmermsft) [inline eng/versioning/version_data_java_files.txt:6]: What is still using this? We've removed this from our source code.

Comment (billwert) [reply eng/versioning/version_data_java_files.txt:6]: Good catch. I'm also not convinced this is useful anymore.. the comment in `update_versions.py` points to #7106 which has been closed for a while. @conniey Should we do something about this and get rid of this text file entirely at some point?

**Pull request merged**